pkg/gateway/routeutils/utils.go

```
@@ -0,0 +1,65 @@
+package routeutils
+
+import (
+	"context"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ListL4Routes retrieves all Layer 4 routes (TCP, UDP, TLS) from the cluster.
+func ListL4Routes(ctx context.Context, k8sClient client.Client) ([]preLoadRouteDescriptor, error) {
+	l4Routes := make([]preLoadRouteDescriptor, 0)
+	tcpRoutes, _ := ListTCPRoutes(ctx, k8sClient)
+	l4Routes = append(l4Routes, tcpRoutes...)
+	udpRoutes, _ := ListUDPRoutes(ctx, k8sClient)
+	l4Routes = append(l4Routes, udpRoutes...)
+	tlsRoutes, _ := ListTLSRoutes(ctx, k8sClient)
+	l4Routes = append(l4Routes, tlsRoutes...)
+	return l4Routes, nil
+}
+
+// ListL7Routes retrieves all Layer 7 routes (HTTP, gRPC) from the cluster.
+func ListL7Routes(ctx context.Context, k8sClient client.Client) ([]preLoadRouteDescriptor, error) {
+	l7Routes := make([]preLoadRouteDescriptor, 0)
```

Same, please log or return the errors. Silent failures are tough to debug.